packages/fetch/src/modifiers.ts

```diff
@@ -0,0 +1,204 @@
+  RequestContext<Result, FetcherArgs>["fetcher"],
+  undefined
+>;
+
+export const wrapFetcher = <Result extends unknown, FetcherArgs extends any[]>(
+  requestContext: RequestContext<Result, FetcherArgs>,
+  wrapper: (originalFetcher: Fetcher<Result, FetcherArgs>) => Fetcher<Result, FetcherArgs>
+) => {
+  const originalFetcher = requestContext.fetcher;
+  if (!originalFetcher) {
+    throw new Error("could not read resource fetcher");
+  }
+  requestContext.fetcher = wrapper(originalFetcher) || originalFetcher;
+};
+
+export const wrapResource = <Result, FetcherArgs>(
+  requestContext: RequestContext<Result, FetcherArgs>,
+  wrapper: (
+    requestContext: RequestContext<Result, FetcherArgs>
+  ) => [props?: { [key: string]: any }, actions?: { [key: string]: any }]
+) => {
+  if (!requestContext.resource) {
+    throw new Error("could not read resource");
+  }
+  const [props, actions] = wrapper(requestContext);
+  props && Object.defineProperties(requestContext.resource[0], props);
+  actions && Object.assign(requestContext.resource[1], actions);
+};
+
+export const withAbort: RequestModifier =
+  <Result extends unknown, FetcherArgs extends any[]>() =>
+  (requestContext: RequestContext<Result, FetcherArgs>) => {
+    wrapFetcher(
+      requestContext,
+      originalFetcher => (requestData: FetcherArgs, info: ResourceFetcherInfo<Result>) => {
+        if (requestContext.abortController) {
+          requestContext.abortController.abort();
+        }
+        requestContext.abortController = new AbortController();
+        const lastRequestDataObj = (() => {
+          for (let l = requestData.length - 1; l >= 0; l--) {
+            if (typeof requestData[l] === "object") {
+              return requestData[l];
+            }
+          }
+          const obj = {};
+          requestData.push(obj);
+          return obj;
+        })();
+        lastRequestDataObj.signal = requestContext.abortController.signal;
+        return originalFetcher(
+          requestData as FetcherArgs,
+          info as ResourceFetcherInfo<Result>
+        ).catch(err => {
+          if (info.value && err.name === "AbortError") {
+            return Promise.resolve(info.value);
+          }
+          throw err;
+        });
+      }
+    );
+    requestContext.wrapResource();
+    wrapResource(requestContext, requestContext => [
+      {
+        aborted: { get: () => requestContext.abortController?.signal.aborted || false },
+        status: { get: () => requestContext.response?.status },
+        response: { get: () => requestContext.response }
+      },
+      {
+        abort: () => requestContext.abortController?.abort()
+      }
+    ]);
+  };
+
+export const withTimeout: RequestModifier = (timeout: number) => requestContext => {
+  wrapFetcher(
+    requestContext,
+    originalFetcher => (requestData, info) =>
+      new Promise((resolve, reject) => {
+        window.setTimeout(() => {
+          // @ts-ignore
```

Comment:
```suggestion
```
It doesn't seem to be needed. It's probably better to use @ts-expect-error to not over ignore stuff.

Reply:
That's a remnant from before I fixed the types.